sim.py

```diff
@@ -142,10 +142,20 @@ def main():
         '--covenant', help='indicate covenant build for output.', choices=config["covenants"])
     parser.add_argument(
         '--local', help='indicate if the simulation should run local.', action='store_true')
+    parser.add_argument(
+        '--auto_download', help='indicate if we should automatically download latest simc.', action='store_true')
     args = parser.parse_args()
 
     sys.path.insert(0, args.dir)
 
+
+    # Download simc if needed
+    if(args.local and args.auto_download):
+        from internal.simc import download_latest
+        from local_secrets import simc_path
+        simc_path['latest'] = download_latest()
```

Comment:
can we move `download_latest()` and some of the helper functions over to an `internal/` script? This file is already kind of bloated so if we can keep things as separate as possible that would be helpful from a logical standpoint.

Reply:
Updated this to pull from internal.auto_download instead.